File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int one = 0, two = 0;

        for (int i = 2; i < cost.length + 1; i++) {
            int temp = Math.min(one + cost[i - 1], two + cost[i - 2]);
            two = one;
            one = temp;
        }

        return one;
    }
}

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int first = 0;
        int second = 0;

        for (int i = 2; i <= cost.size(); i++) {
            int temp = min(first + cost[i - 2], second + cost[i - 1]);
            first = second;
            second = temp;
        }

        return second;
    }
};